design/overview.md

```diff
@@ -0,0 +1,145 @@
+# Design Overview
+
+Details the overall design of Formica - a management layer over fleet
+
+- _Formica_ is a genus of ants, see https://en.wikipedia.org/wiki/Formica
+- See overview.png for a more graphical view of this material
+
+## General Design Overview
+- Service running inside cluster named ```formicad```
+- Command line tool to interact with ```formicad``` named ```formicactl```
+
+### ```formicad```
+- Handles generating unit files from unit file definitions
+- Handles submitting and updating unit files
+- Supports submitting unit file definitions
+- Supports querying against the state of the unit file definition being processed
```

Comment:
My ideas were largely:
- Has this unit file definition been parsed into unit files?
- How many of the unit files have been submitted?
As we mentioned in our chat, it's essentially querying the task system for the state of processing

The aim of this querying is to allow clients to see the state of processing, and these are the two main responsibilities of ```formicad```